pandas/tests/frame/test_constructors.py

```diff
@@ -2698,6 +2698,14 @@ def test_frame_from_dict_with_mixed_tzaware_indexes(self):
 
 
 class TestDataFrameConstructorWithDtypeCoercion:
+    def test_none_values_on_string_columns(self):
+        # Creating DataFrame with column "a" as string and a None value
+        data = {"a": ["1", "2", None]}
+        df = DataFrame(data=data, dtype="str")
+
+        # Checking that the value of the third row in column "a" is actually None
+        assert isna(df.loc[2, "a"])
```

Comment:
```suggestion
        assert df.loc[2, "a"] is None
```

Reply:
Greetings @mroeschke, i made the changes as requested